sharedresource/v1alpha1/types_share.go

```diff
@@ -0,0 +1,23 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SharedResourceStatus contains the observed status of shared resource. Read-only.
+type SharedResourceStatus struct {
+	// conditions represents any observations made on this particular shared resource by the underlying CSI driver or Share controller.
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+// ResourceReference represents the ConfigMap or Secret for this share
+type ResourceReference struct {
+	// name represents the name of the ConfigMap or Secret that is being referenced.
```

Comment:
This doc is incorrect.  When in a ConfigMap it's only a ConfigMap, when it is a Secret it must be a secret.  

Having distinct types allows distinct doc.  See the recommendation to specify by type in the ObjectReference godoc.